### DISCUSSION TITLE
Calling a custom page in a modal window from a custom table action

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2462] [Help] [answered]
Hi all,

I would like to call a custom view (based on a custom page) inside a modal window by using a custom table action.

I am aware that it is possible to have ->form([]) added to an action and then define a form in it using the entire form builder capabilities.

Is there also a possibility to call a custom page (not "only" the form) in a modal window from a table action - and if so is it possible to control the size of the modal window?

Thanks!

**Comment by danharrin** [comment, marked as answer]
There is a `modalWidth()` method on the action

And you can pass in a custom view to `modalContent()`

**Reply by vv-ac** [reply]
Thank you, I did not know about this one, will likely try out today.

Is this also applicable to simple resources that are created in modal windows by default (to be then applied somwhere in ManageResources.php)?

**Reply by danharrin** [reply]
Yeah it works for all actions

Tbh, it would be nice if we could just specify a ->modalContent() and pass a view without using a form...

**Reply by ghost** [reply]
Hello @danharrin how do you call modalContent ? I don't see it neither in doc nor in source code. Thanks.

**Reply by danharrin** [reply]
My reply was a suggestion that someone can PR modalContent() in the future

**Comment by danharrin** [comment]
`modalContent()` is now available, to pass a custom view to an action
